public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -79,17 +79,24 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
       })
       .pipe(
         map(({ data }) => {
-          const results = data.results['annotationQuery'].tables[0].rows.map((v: any) => {
-            return {
-              annotation: annotation,
-              time: Date.parse(v[0]),
-              title: v[1],
-              tags: [],
-              text: v[3],
-            } as any;
+          const dataQueryResponse = toDataQueryResponse({
+            data: data,
           });
-
-          return results;
+          const df: any = [];
+          if (dataQueryResponse.data.length !== 0) {
+            for (let i = 0; i < dataQueryResponse.data.length; i++) {
+              for (let j = 0; j < dataQueryResponse.data.length; j++) {
```

Comment:
These two loop over the same index, so we're missing out on some events. Think it needs to be something like this:
```suggestion
              for (let j = 0; j < dataQueryResponse.data[i].length; j++) {
```

Reply:
Fixed seconds before this!! 😁